cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java

```diff
@@ -578,10 +582,60 @@ private Future<Void> pauseResume(Reconciliation reconciliation, String host, Kaf
         }
     }
 
+    private  Future<ConnectorStatusAndConditions> autoRestartFailedConnectorAndTasks(Reconciliation reconciliation, String host, KafkaConnectApi apiClient, String connectorName, KafkaConnectorSpec connectorSpec, ConnectorStatusAndConditions status, CustomResource resource) {
+        JsonObject statusResultJson = new JsonObject(status.statusResult);
+        if (connectorSpec.getAutoRestart().isEnabled() && (connectorHasFailed(statusResultJson) || (failedTaskIds(statusResultJson)).size() > 0)) {
+            return getPreviousKafkaConnectorStatus(connectorOperator, resource)
+                 .compose(previousStatus -> {
+                     if (previousStatus == null) {
+                         previousStatus = new KafkaConnectorStatus();
+                     }
+                     return Future.succeededFuture(previousStatus.getAutoRestart());
+                 })
+                 .compose(previousAutoRestartStatus -> {
+                     if (shouldAutoRestart(previousAutoRestartStatus)) {
+                         LOGGER.debugCr(reconciliation, "Auto restarting connector {}", connectorName);
+                         return apiClient.restart(host, port, connectorName, true, true)
+                             .compose(
+                                 success -> Future.succeededFuture(status),
+                                 throwable -> {
+                                     status.autoRestart = StatusUtils.incrementAutoRestartStatus(previousAutoRestartStatus);
+                                     String message = "Failed to auto restart for the " + status.autoRestart.getCount() + " time(s) connector  " + connectorName + ". " + throwable.getMessage();
+                                     LOGGER.warnCr(reconciliation, message);
+                                     return Future.succeededFuture(status);
+                                 });
+                     } else {
+                         // keep previous auto restart status
+                         status.autoRestart = previousAutoRestartStatus;
+                         return Future.succeededFuture(status);
+                     }
+                 });
+        } else {
+            return Future.succeededFuture(status);
+        }
+    }
+
+    /**
+     * Connector or tasks failed should only restart at minute 0, 0, 2, 6, 12, 20 and 30
```

Comment:
Is the double 0 (`0, 0`) just a typo? Or will it actually restart for some reason twice immediately? I guess it would based on the algorithm.

Reply:
I did it to restart at next reconciliation not matter how much time past between them, but it was not in the proposal I will revert this.